In [1]:
import pandas as pd
import numpy as np
import os
import seaborn as sns
import matplotlib.pyplot as plt



In [2]:
df = pd.read_csv('/Users/majapedigo/Desktop/ Capstone Project II/Game_sales.csv')


In [3]:
df.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [4]:
df.info(memory_usage='deep')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 5.2 MB


In [5]:
df.nunique()

Rank            16598
Name            11493
Platform           31
Year               39
Genre              12
Publisher         578
NA_Sales          409
EU_Sales          305
JP_Sales          244
Other_Sales       157
Global_Sales      623
dtype: int64

Info tells us that there are many null values in the 'year'column, some are missing in the 'Publisher' column, and 'year' should be converted to int type.
'Name' column shows that there are games that were ranked in multiple places across the platforms and over the years.

In [6]:
df.apply(pd.unique)

Rank            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...
Name            [Wii Sports, Super Mario Bros., Mario Kart Wii...
Platform        [Wii, NES, GB, DS, X360, PS3, PS2, SNES, GBA, ...
Year            [2006.0, 1985.0, 2008.0, 2009.0, 1996.0, 1989....
Genre           [Sports, Platform, Racing, Role-Playing, Puzzl...
Publisher       [Nintendo, Microsoft Game Studios, Take-Two In...
NA_Sales        [41.49, 29.08, 15.85, 15.75, 11.27, 23.2, 11.3...
EU_Sales        [29.02, 3.58, 12.88, 11.01, 8.89, 2.26, 9.23, ...
JP_Sales        [3.77, 6.81, 3.79, 3.28, 10.22, 4.22, 6.5, 2.9...
Other_Sales     [8.46, 0.77, 3.31, 2.96, 1.0, 0.58, 2.9, 2.85,...
Global_Sales    [82.74, 40.24, 35.82, 33.0, 31.37, 30.26, 30.0...
dtype: object

In [7]:
df['Year'].value_counts(dropna=False)

2009.0    1431
2008.0    1428
2010.0    1259
2007.0    1202
2011.0    1139
2006.0    1008
2005.0     941
2002.0     829
2003.0     775
2004.0     763
2012.0     657
2015.0     614
2014.0     582
2013.0     546
2001.0     482
1998.0     379
2000.0     349
2016.0     344
1999.0     338
1997.0     289
NaN        271
1996.0     263
1995.0     219
1994.0     121
1993.0      60
1981.0      46
1992.0      43
1991.0      41
1982.0      36
1986.0      21
1989.0      17
1983.0      17
1990.0      16
1987.0      16
1988.0      15
1985.0      14
1984.0      14
1980.0       9
2017.0       3
2020.0       1
Name: Year, dtype: int64

In [8]:
df.shape

(16598, 11)

271 NaN values in the column 'Year', years 2017, 2020, and 1980 have only a few rows so we will drop them. 

In [9]:
df_drop_index = df[df.Year.isin([2017, 2020, 1980])].index
df = df.drop(df_drop_index)
df.shape

(16585, 11)

In [10]:
missing_count = df.shape[0] - df.count() 
print(missing_count)

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64


In [11]:
## Since the data contain time-trend data, Year-null value shoudl be dropped ## 

df = df.dropna(subset=['Year'])
df.isnull().sum()

Rank             0
Name             0
Platform         0
Year             0
Genre            0
Publisher       36
NA_Sales         0
EU_Sales         0
JP_Sales         0
Other_Sales      0
Global_Sales     0
dtype: int64

In [12]:
df.shape

(16314, 11)

We should also change the 'Year' type to int

In [13]:
df = df.astype({"Year": int})

In [14]:
from scipy.stats import f_oneway
 
# Running the one-way anova test between Publisher and Global_Sales
# Assumption(H0) is that Publisher and Global_Sales are NOT correlated
 
# Finds out the Global_Sales data for each Publisher as a list
CategoryGroupLists=df.groupby('Publisher')['Global_Sales'].apply(list)
 
# Performing the ANOVA test
# We accept the Assumption(H0) only when P-Value &gt; 0.05
AnovaResults = f_oneway(*CategoryGroupLists)
print('P-Value for Anova is: ', AnovaResults[1])

P-Value for Anova is:  4.787429646654068e-136



Performed Anova test to check correlation between a categorical variable Publisher and target numerical variable, as Publisher variable has many missing values and we are thinking about removing it completely, as it may not be relevant for our prediction.
However, Anova test showed P-value to be less than 0.05, and this means that Publisher and Global_Sales ARE correlated, and Publisher cannot be removed. 

However, since only 58 rows have missing values, compared to the total number of rows this is not a lot, so we will drop those rows as well.



In [15]:
df = df.dropna(subset=['Publisher'])
df.isnull().sum()

Rank            0
Name            0
Platform        0
Year            0
Genre           0
Publisher       0
NA_Sales        0
EU_Sales        0
JP_Sales        0
Other_Sales     0
Global_Sales    0
dtype: int64

In [16]:
df[df.duplicated(keep=False)]

Empty DataFrame
Columns: [Rank, Name, Platform, Year, Genre, Publisher, NA_Sales, EU_Sales, JP_Sales, Other_Sales, Global_Sales]
Index: []

No duplicated rows in this dataset.


In [17]:
# rename  columns to lower case
df.columns= df.columns.str.lower()
df.columns

Index(['rank', 'name', 'platform', 'year', 'genre', 'publisher', 'na_sales',
       'eu_sales', 'jp_sales', 'other_sales', 'global_sales'],
      dtype='object')

In [18]:
df.describe()

rank          year      na_sales      eu_sales      jp_sales  \
count  16278.000000  16278.000000  16278.000000  16278.000000  16278.000000   
mean    8291.720482   2006.417373      0.265192      0.147808      0.078893   
std     4791.725737      5.798835      0.821982      0.509495      0.311997   
min        1.000000   1981.000000      0.000000      0.000000      0.000000   
25%     4135.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8295.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12439.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2016.000000     41.490000     29.020000     10.220000   

        other_sales  global_sales  
count  16278.000000  16278.000000  
mean       0.048456      0.540622  
std        0.190155      1.567561  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.480000  
max       10.570000     82.740000

Our sales columns do not have outliers according to the above.